finesse/hardware/temperature/dp9800.py

```diff
@@ -161,13 +161,13 @@ def read(self) -> bytes:
         Raises:
             DP9800Error: Malformed message received from device
         """
-        # require at least 4 bytes else checks will fail
-        min_length = 4
         try:
-            data = self.serial.read_until(b"\x00", size=min_length)
+            data = self.serial.read_until(b"\x00")
```

Comment:
Why has this been changed? There's no info in the commit message about why this change was needed.

Reply:
[read_until()](https://pyserial.readthedocs.io/en/latest/pyserial_api.html#serial.Serial.read_until) reads until the expected sequence, in this case `b"\x00"`, has been found, unless `size` bytes have been read, at which point it stops reading.

When reading from the actual device, reading would stop once `min_length` bytes had been read. This is set to 4 bytes, as this is the minimum that is required for the integrity checks to work, but we want to read the entire message, not just the first 4 bytes.

Reply:
Ahh I see. That makes sense.